### ISSUE TITLE
build  warning: unqualified call to 'std::move' 

### ISSUE THREAD

**Issue by calvin2021y** [Tencent/MMKV#1077]
## Note:

improve the warning message

### The language of MMKV

c++

### The version of MMKV

v1.2.16

### The platform of MMKV

linux

### What's the issue?

```sh
mmkv/Core/MiniPBCoder.cpp:243:21: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
        v.push_back(move(value));
                    ^
                    std::
mmkv/Core/MiniPBCoder.cpp:262:39: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
                    dictionary[key] = move(kvHolder);
                                      ^
                                      std::
[ 58%] Building CXX object CMakeFiles/core.dir/MemoryFile_Android.cpp.o
[ 61%] Building CXX object CMakeFiles/core.dir/MemoryFile_Linux.cpp.o
[ 63%] Building CXX object CMakeFiles/core.dir/MemoryFile_Win32.cpp.o
mmkv/Core/MiniPBCoder.cpp:305:39: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
                    dictionary[key] = move(kvHolder);
                                      ^
                                      std::
[ 66%] Building CXX object CMakeFiles/core.dir/MemoryFile_OSX.cpp.o
[ 69%] Building CXX object CMakeFiles/core.dir/ThreadLock.cpp.o
[ 72%] Building CXX object CMakeFiles/core.dir/ThreadLock_Win32.cpp.o
mmkv/Core/MMKV_IO.cpp:343:22: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return make_pair(move(buffer), totalSize);
                     ^
                     std::
mmkv/Core/MMKV_IO.cpp:385:32: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
        return doFullWriteBack(move(preparedData), nullptr);
                               ^
                               std::
mmkv/Core/MMKV_IO.cpp:533:31: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
                itr->second = move(kvHolder);
                              ^
                              std::
mmkv/Core/MMKV_IO.cpp:535:51: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
                itr->second = KeyValueHolderCrypt(move(data));
                                                  ^
                                                  std::
mmkv/Core/MMKV_IO.cpp:549:62: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
                m_dicCrypt->emplace(key, KeyValueHolderCrypt(move(data)));
                                                             ^
                                                             std::
mmkv/Core/MMKV_IO.cpp:745:36: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
            return doFullWriteBack(move(preparedData), newCrypter);
                                   ^
                                   std::
mmkv/Core/MMKV.cpp:428:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:440:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:452:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:464:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:476:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:488:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:500:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:534:26: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    return setDataForKey(move(data), key);
                         ^
                         std::
mmkv/Core/MMKV.cpp:566:22: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
            result = move(input.readData());
                     ^
                     std::
[ 77%] Building CXX object CMakeFiles/core.dir/aes/AESCrypt.cpp.o
[ 77%] Building CXX object CMakeFiles/core.dir/aes/openssl/openssl_aes_core.cpp.o
[ 80%] Building CXX object CMakeFiles/core.dir/aes/openssl/openssl_cfb128.cpp.o
mmkv/Core/MMKV.cpp:1151:21: warning: unqualified call to 'std::move' [-Wunqualified-std-cast-call]
    File dstCRCFile(move(dstCRCPath), OpenFlag::ReadWrite | OpenFlag::Create);
```

**Comment by lingol** [comment]
Please provide the compiler version you are using.

**Comment by calvin2021y** [comment]
clang-15

**Comment by lingol** [comment]
Fixed in v1.3.0.
